gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -856,74 +983,104 @@ gst_tensor_filter_common_set_property (GstTensorFilterPrivate * priv,
-      g_free_const (prop->custom_properties);
-      prop->custom_properties = g_value_dup_string (value);
+      if (!priv->prop.fw_opened) {
+        g_free_const (prop->custom_properties);
+        prop->custom_properties = g_value_dup_string (value);
+      } else {
+        /** TODO: update the properties */
+      }
+
       g_debug ("Custom Option = %s\n", prop->custom_properties);
       break;
     case PROP_ACCELERATOR:
     {
-      /**
-       * TODO: allow updating the subplugin accelerator after it has been init
-       * by reopening
-       */
+      gchar *accelerators = g_value_dup_string (value);
       if (priv->prop.fw_opened == TRUE) {
+        /** TODO: update the properties */
+        g_free (accelerators);
         break;
       }
 
-      /** TODO: set this property based on the framework version V0 or V1 */
-      prop->accl_str = g_value_dup_string (value);
+      if (GST_TF_FW_V1 (priv->fw)) {
+        gst_tensor_filter_parse_accelerator (priv, &priv->prop, accelerators);
+        g_free (accelerators);
+      } else {
+        prop->accl_str = accelerators;
+      }
       break;
     }
     case PROP_IS_UPDATABLE:
     {
-      if (priv->fw->reloadModel != NULL)
-        priv->is_updatable = g_value_get_boolean (value);
+      if (GST_TF_FW_V0 (priv->fw) && priv->fw->reloadModel == NULL) {
+        break;
+      } else if (GST_TF_FW_V1 (priv->fw) &&
+          priv->fw->eventHandler (prop, priv->privateData, RELOAD_MODEL, NULL)
+          == -ENOENT) {
+        break;
+      }
+
+      priv->is_updatable = g_value_get_boolean (value);
       break;
     }
     case PROP_INPUTLAYOUT:
     {
-      guint num_layouts;
-#if 0 /** @todo allow updating the data layout after fw has been opened */
-      if (priv->prop.fw_opened == TRUE) {
-        g_warning
-            ("Updating data layout is not supported after opened framework.");
-        break;
-      }
-#endif
-      num_layouts = gst_tensors_parse_layouts_string (prop->input_layout,
-          g_value_get_string (value));
+      if (!prop->output_configured && value) {
+        guint num_layouts;
 
-      if (prop->input_meta.num_tensors > 0 &&
-          prop->input_meta.num_tensors != num_layouts) {
-        g_warning ("Invalid input-layout, given param does not fit.");
-      }
+        num_layouts = gst_tensors_parse_layouts_string (prop->input_layout,
+            g_value_get_string (value));
 
-      prop->input_meta.num_tensors = num_layouts;
+        if (prop->input_meta.num_tensors > 0 &&
+            prop->input_meta.num_tensors != num_layouts) {
+          g_warning ("Invalid input-layout, given param does not fit.");
+        }
+
+        prop->input_meta.num_tensors = num_layouts;
+      } else if (value) {
+        /** Update the properties */
+        if (GST_TF_FW_V0 (priv->fw)) {
+          /* Once configured, it cannot be changed in runtime */
+          g_assert (!prop->input_configured);
```

Comment:
No. 
This does not incur any GST-CAP renegotiation. You may chage it in runtime w/ V0, I suppose.

Reply:
This follows the same issue as custom properties.
How will the framework know if the layout has been changed for V0?

Reply:
ah.. yes, that's right. Let's prohibit it for V0.